LiteFarmOrg/LiteFarm
# Get farms by offset range endpoint

This PR creates an endpoint that, when hit, pulls the farm id's of all farms within a certain UTC offset range.

To test:

1. Run `npm run migrate:testing:db` and `npm run migrate:testing:db` in the api folder
2. Go into the 'farm' table your database and modify the 'utc_offset' field of the farms. Have different values so you can properly test for range.
3. To receive a response from this endpoint we need to authenticate via a token. First go into 'farmRoute.js' and comment out lines 68 & 69. Then launch your local Litfarm environment, open developer tools, go to application, and copy the value for 'id_token'
4. Now we will make a request. Open a terminal and write `curl -H "Authorization: Bearer <YOUR-ID-TOKEN>" http://localhost:5001/farm/utc_offset_by_range/<CUSTOM-MIN-VALUE>/<CUSTOM-MAX-VALUE>` (Make sure to replace the values appropriately)

You should receive a list of the farm id's of all the farms that match your specification.
